bigquery_etl/schema/__init__.py

```diff
@@ -49,7 +49,7 @@ def for_table(cls, project, dataset, table, partitioned_by=None):
         query = f"SELECT * FROM {project}.{dataset}.{table}"
 
         if partitioned_by:
-            query += f" WHERE {partitioned_by} = DATE('2020-01-01')"
+            query += f" WHERE DATE({partitioned_by}) = DATE('2020-01-01')"
```

Comment:
I was playing around with schema validation here. Ran into this bug that causes the query to fail if `partitioned_by` is a `TIMESTAMP`

Reply:
If `partitioned_by` is already a DATE, this may fail. Can you verify that this is compatible with both dates and timestamps?

Reply:
For `clients_daily_v6` the `partitioned_by` is a `DATE` and it's running fine.

Generally, converting a `DATE` to a `DATE` shouldn't cause any problems I can think of:

```sql
WITH ts_data AS (
  SELECT TIMESTAMP('2021-03-01 11:12:22') as ts
)
SELECT ts from ts_data
WHERE DATE(DATE(DATE(ts))) > DATE('2021-01-01')
```